tests/ui/lint/unused/issue-96606.rs

```diff
@@ -0,0 +1,7 @@
+#[deny(unused)]
+fn main() {
+    let arr = [0; 10];
+    let _ = arr[(0)]; //~ ERROR unnecessary parentheses around index expression
+    let _ = arr[{0}]; //~ ERROR unnecessary braces around index expression
```

Comment:
```suggestion
    let _ = arr[{0}]; //~ ERROR unnecessary braces around index expression
    let _ = arr[{ let x = 0; x }];
```
here we shouldn't lint on the braces, please also add that as a test.